test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/CpsPackageReferenceProjectTests.cs

```diff
@@ -3251,6 +3252,31 @@ public async Task GetTransitivePackagesAsync_WithTransitivePackageReferences_Ret
             }
         }
 
+        [Fact]
+        public async Task GetPackageSpecsAndAdditionalMessagesAsync_NoRestoreInfoInSolutionManager_ThrowsProjectNotNominatedException()
+        {
+            // Arrange
+            var projectSystemCache = new Mock<IProjectSystemCache>();
+            var unconfiguredProject = new Mock<UnconfiguredProject>();
```

Comment:
Here is how we're controlling nomination?
Is it because of `UnconfiguredProject` type?

Reply:
the project system cache (declared above) is empty, hence no nomination.